src/components/ContentSidebar/ActivityFeed/approval-comment-form/ApprovalCommentForm.js

```diff
@@ -86,7 +86,7 @@ class ApprovalCommentForm extends React.Component<Props, State> {
             return;
         }
 
-        if (formData.addApproval === 'on') {
+        if (formData.addApproval === 'on' && createTask) {
```

Comment:
Shouldn't we not be rendering task related stuff if we don't have createTask callback? This form data would never exist in that case. Same goes for other two checks you added.

Reply:
True. We don't render them but since they're optional props and it's possible that you have only one set of methods available to you (task OR comment). Since some of the methods are optional which could trigger this method to be called, we need to account for them in the methods.